sources/packages/backend/apps/api/src/route-controllers/institution-locations/institution-location.institutions.controller.ts

```diff
@@ -265,4 +266,41 @@ export class InstitutionLocationInstitutionsController extends BaseController {
       };
     });
   }
+
+  /**
+   * Gets all eligible application that can be requested for application
+   * offering change.
+   * @param locationId location id.
+   * @param paginationOptions options to execute the pagination.
+   * @returns list of eligible application that can be requested for
+   * application offering change.
+   */
+  @HasLocationAccess("locationId")
+  @Get(":locationId/active-applications/request-change")
```

Comment:
The endpoint seems to be too generic to represent the offering change.
Should we start a new controller specific to the entire feature?
For instance:

@Get(":locationId/offering-change-request/[available,InProgress,etc]") - Institution/Ministry
@Get(":locationId/offering-change-request/") - Ministry
@Get(":locationId/offering-change-request/:offeringChangeRequestId") - Institution/Ministry
@Post(":locationId/offering-change-request") - Institution only
@Patch(":locationId/offering-change-request") - Ministry/Student

The above endpoints do not represent what we actually need, they were added to help to demonstrate the point that we will need a variety of endpoints for Institution/Ministry/Students, and a new folder under the `route-controllers` seems to be more reasonable. Thoughts

Reply:
I think it's a good idea to have a separare controller for "offering change request".

Reply:
I agree that having a seperate controller will make it better to organize the endpoints for the feature.
Having said that, for the endpoints, using offering-change-request will be more precise from API side.